fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -47,6 +47,50 @@ pub fn from_url_to_url_auth(url: &Url) -> Result<(String, Auth)> {
         }),
         if url.username().is_empty() {
             Auth::None
+        } else if url.password().is_none() {
+            if let Some(cookie_file) = rpc_cookie_file {
+                let cookie = std::fs::read_to_string(cookie_file)?;
+                let mut auth = Auth::None;
+                for line in cookie.lines() {
+                    let tokens: Vec<_> = line.split('=').collect();
+                    if tokens.len() == 2 {
+                        match tokens[0] {
+                            "rpcuser" => {
```

Comment:
I thought the format of cookie file was just one line with `<user>:<password>` or something like that?